internal/pkg/telemetry/metric/prometheus/prometheus.go

```diff
@@ -88,6 +89,43 @@ func ServeMetrics(ctx context.Context, serviceName, listenAddr string, logger lo
 	}
 
 	// Create OpenTelemetry metrics provider
-	provider := metric.NewMeterProvider(metric.WithReader(exporter), metric.WithResource(res))
+	provider := metric.NewMeterProvider(
+		metric.WithReader(exporter),
+		metric.WithResource(res),
+		metric.WithView(View()),
+	)
 	return provider, nil
 }
+
+func View() metric.View {
+	ignoreAttrs := metric.NewView(
+		metric.Instrument{Name: "*"},
+		metric.Stream{AttributeFilter: func(value attribute.KeyValue) bool {
+			switch value.Key {
+			// Remove invalid otelhttp metric attributes with high cardinality.
+			// https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3765
+			case "net.sock.peer.addr",
+				"net.sock.peer.port",
+				"http.user_agent",
+				"http.client_ip",
+				"http.request_content_length",
+				"http.response_content_length":
+				return false
```

Comment:
`otelhttp` ma bug a generuje dimenzie metrik s velkym mnozstvom moznosti, napr. `client_ip`.
Kazda komponacia atributov sa v DD pocita ako samostatna metrika a podla toho sa plati.
Aj interne v Prometheus sa to tak uklada, .... 

TLDR: je to nespravne, mali by sme tolko metrik kolko mame uzivatelov/IP adresie. Podobne je to aj s dalsimi uvedenými attributes.